darts/models/forecasting/tide_model.py

```diff
@@ -0,0 +1,609 @@
+        if self.future_cov_dim > 0:
+            x_dynamic_covariates = torch.cat(
+                [
+                    x_future_covariates,
+                    x[
+                        :,
+                        :,
+                        None if self.future_cov_dim == 0 else -self.future_cov_dim :,
+                    ],
+                ],
+                dim=1,
+            )
+
+            # project input features across all input time steps
+            x_dynamic_covariates_proj = self.feature_projection(x_dynamic_covariates)
+
+        else:
+            x_dynamic_covariates = None
+            x_dynamic_covariates_proj = None
+
+        # extract past covariates, if they exist
+        if self.input_dim - self.output_dim - self.future_cov_dim > 0:
+            x_past_covariates = x[
+                :,
+                :,
+                self.output_dim : None
+                if self.future_cov_dim == 0
+                else -self.future_cov_dim :,
+            ]
+        else:
+            x_past_covariates = None
+
+        # setup input to encoder
+        encoded = [
+            x_lookback,
+            x_past_covariates,
+            x_dynamic_covariates_proj,
+            x_static_covariates,
+        ]
+        encoded = [t.flatten(start_dim=1) for t in encoded if t is not None]
+        encoded = torch.cat(encoded, dim=1)
+
+        # encoder, decode, reshape
+        encoded = self.encoders(encoded)
+        decoded = self.decoders(encoded)
+
+        # get view that is batch size x output chunk length x self.decoder_output_dim x nr params
+        decoded = decoded.view(x.shape[0], self.output_chunk_length, -1)
+
+        # stack and temporally decode with future covariate last output steps
+        temporal_decoder_input = [
+            decoded,
+            x_dynamic_covariates_proj[:, -self.output_chunk_length :, :]
+            if self.future_cov_dim > 0
+            else None,
+        ]
+        temporal_decoder_input = [t for t in temporal_decoder_input if t is not None]
+
+        temporal_decoder_input = torch.cat(temporal_decoder_input, dim=2)
+        temporal_decoded = self.temporal_decoder(temporal_decoder_input)
+
+        # pass x_lookback through self.lookback_skip but swap the last two dimensions
+        # this is needed because the skip connection is applied across the input time steps
+        # and not across the output time steps
+        skip = self.lookback_skip(x_lookback.transpose(1, 2)).transpose(1, 2)
+
+        # add skip connection
+        y = temporal_decoded + skip.reshape_as(
+            temporal_decoded
+        )  # skip.view(temporal_decoded.shape)
+
+        y = y.view(-1, self.output_chunk_length, self.output_dim, self.nr_params)
+
+        if self.use_reversible_instance_norm:
+            for i in range(self.nr_params):
+                y[:, :, :, i] = self.rin.inverse(y[:, :, :, i])
```

Comment:
we can avoid iterating over nr_params if we expand the dimension of `stdev`, `mean`, `affine_weight`, `affine_bias` in `RINorm.inverse` (all models should have output shape `(batch_size, output_chunk_length, n_targets, nr_params)`).

just as an example:
```
x = x * self.stdev.view(self.stdev.shape + (1,))
```

then we can call `inverse` directly:
```suggestion
            y = self.rin.inverse(y)
```